### DISCUSSION TITLE
ESP32-WROOM-32U: Is bad signal strength expected without external antenna?

### DISCUSSION THREAD

**Discussion by retroheaven** [darthcloud/BlueRetro#115] [Q&A] [answered]
Can anyone confirm whether ESP32-WROOM-32U has very bad bluetooth signal if not attaching an external antenna? I barely get the connection working when using the device with N64. If i'm close to 50 cm from the device it kind of works, but it is not 100 % stable even then. Same instability noticed when connecting to the device via blueretro.io. Lots of dropped connections.

**Comment by darthcloud** [comment, marked as answer]
No need to open 2 topic for same thing,

Pretty much no one is using external antenna.

Its wireless, so it's possible your specific location got a lot of noise. (Like in dense appartement complex) 

Also It's possible you got a single device making interference to an otherwise perfect environment. Cheap wireless video surveillance/baby camera are notorious for this.

Remove the 3.3v connection from n64, Keep only data and GND. Power from USB. Is it better that way?

You could also have a rogue Bluetooth device always trying to connect to Blueretro.
With the USB power setup mentioned previously, Try to get the logs maybe I can spot something in it.
https://github.com/darthcloud/BlueRetro/wiki/Getting-BlueRetro-debug-logs-via-Serial-port-Windows-10

**Reply by retroheaven** [reply]
Thank you for helping out and giving a thorough reply with suggestions. I'm sorry, but I couldn't find any other topic with this same question.

I indeed live in an apartment complex. However, I am using the Blueretro via the PS1UnoRetro as well and do not experience the same issue with it.

I now detached the 3.3v connection and tried again. Got the same result. See the attached log.

[esp32-wroom-32u.txt](https://github.com/darthcloud/BlueRetro/files/7698517/esp32-wroom-32u.txt)

**Reply by darthcloud** [reply]
ok I didn't realize you got an ESP without internal antenna. You need to connect an antenna on the U.FL connector for sure.

From the log I see N64 ctrl connect but then disconnect on his own

**Reply by retroheaven** [reply]
Aha, ok. So in hindsight, WROOM-32S is the better choice as long as not external antenna is wanted, right? I didn't see the diff between 32U and 32S before I started.

**Reply by darthcloud** [reply]
Yes

**Reply by retroheaven** [reply]
Thank you! I will try with internal antenna when I get hold of one.

**Reply by retroheaven** [reply]
Just wanted to confirm the obvious: It works much better with a model with built-in antenna :) I got 32D as recommended in Bill of Material.
